pypesto/store/save_to_hdf5.py

```diff
@@ -315,3 +320,110 @@ def write_result(result: Result,
     if sample:
         pypesto_sample_writer = SamplingResultHDF5Writer(filename)
         pypesto_sample_writer.write(result, overwrite=overwrite)
+
+
+def save_objective_infos(
+        objective: Union[AmiciObjective,
+                         AggregatedObjective,
+                         Objective],
+        f: h5py.File
+):
+    """
+    Saves the Objective Infos in a Dictonary.
+
+    Parameters
+    ----------
+    objective:
+        The objective whose information is to be stored.
+    f:
+        The h5py.File in which the infos are stored.
+    """
+    if isinstance(objective, AmiciObjective):
+        save_amici_objective_info(objective, f)
+    elif isinstance(objective, AggregatedObjective):
+        save_aggregated_objective_info(objective, f)
+    else:
+        save_function_objective_info(objective, f)
```

Comment:
if ... else statements are generally bad design in such a context, as they are inflexible to extend. rather, the objective function should have a method to return such information, see e.g. https://github.com/ICB-DCM/pyABC/blob/main/pyabc/distance/distance.py#L106. this function should then be extended/modified in subclasses.